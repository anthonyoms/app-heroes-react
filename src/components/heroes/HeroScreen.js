import React, { useMemo } from "react";
import { useParams, Navigate, useNavigate } from "react-router-dom";
import { getHeroById } from "../../selectors/getHeroById";

export const HeroScreen = ({history}) => {
  const navigate = useNavigate();
  const { heroeId } = useParams();

  const hero = useMemo(() => getHeroById(heroeId), [heroeId]);
  if (!hero) {
    return <Navigate to="/" />;
  }
  const handleReturn = () => {
    console.log(window.history)
    if (window.history <= 2) {
      heroeId.includes("dc")
        ? navigate("/dc")
        : navigate("/marvel");
    } else {
      navigate(-1);
    }
  };
  const { superhero, publisher, alter_ego, first_appearance, characters } =
    hero;
  return (
    <div className="row mt-5">
      <div className="col-4">
        <img
          src={`../assets/heroes/${heroeId}.jpg`}
          alt={superhero}
          className="img-thumbnail card-columns animate__animated animate__fadeInLeft"
        />
      </div>
      <div className="col-8  animate__animated animate__fadeIn">
        <h3>{superhero}</h3>
        <ul className="list-group list-group-flush card-columns">
          <li className="list-group-item">
            <b>Alter ego: </b>
            {alter_ego}
          </li>
          <li className="list-group-item">
            <b>Publisher: </b>
            {publisher}
          </li>
          <li className="list-group-item">
            <b>first appearance: </b>
            {first_appearance}
          </li>
        </ul>

        <h5>Characters: </h5>
        <p>{characters}</p>
        <button className="btn btn-outline-info" onClick={handleReturn}>
          Return
        </button>
      </div>
    </div>
  );
};
