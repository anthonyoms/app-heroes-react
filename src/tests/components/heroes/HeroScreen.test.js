import { mount} from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router";
import { HeroScreen } from "../../../components/heroes/HeroScreen";
import { MarvelScreen } from "../../../components/marvel/MarvelScreen";



describe("Pruebas en <HeroScreen />", () => {
  test("debe de mostrar el componente de redireccion al /", () => {
    const wrapper = mount(
      <MemoryRouter>
        <Routes>
        <Route path="/hero" element={<HeroScreen />} />
        <Route path="/" element={<MarvelScreen />} />
        </Routes>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
