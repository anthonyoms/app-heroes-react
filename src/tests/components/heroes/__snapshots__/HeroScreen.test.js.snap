// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroScreen /> debe de mostrar el componente de redireccion al / 1`] = `""`;
